// n8n-nodes-scraping-dog/dist/nodes/ScrapingDog/resources/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.resources = exports.amazonSearchResource = exports.bingSearchResource = exports.googleNewsResource = exports.googleImagesResource = exports.googleSearchResource = exports.scrapeUrlResource = void 0;
var scrapeUrl_1 = require("./scrapeUrl");
Object.defineProperty(exports, "scrapeUrlResource", { enumerable: true, get: function () { return scrapeUrl_1.scrapeUrlResource; } });
var googleSearch_1 = require("./googleSearch");
Object.defineProperty(exports, "googleSearchResource", { enumerable: true, get: function () { return googleSearch_1.googleSearchResource; } });
var googleImages_1 = require("./googleImages");
Object.defineProperty(exports, "googleImagesResource", { enumerable: true, get: function () { return googleImages_1.googleImagesResource; } });
var googleNews_1 = require("./googleNews");
Object.defineProperty(exports, "googleNewsResource", { enumerable: true, get: function () { return googleNews_1.googleNewsResource; } });
var bingSearch_1 = require("./bingSearch");
Object.defineProperty(exports, "bingSearchResource", { enumerable: true, get: function () { return bingSearch_1.bingSearchResource; } });
var amazonSearch_1 = require("./amazonSearch");
Object.defineProperty(exports, "amazonSearchResource", { enumerable: true, get: function () { return amazonSearch_1.amazonSearchResource; } });
exports.resources = {
    scrapeUrl: 'scrapeUrl',
    googleSearch: 'googleSearch',
    googleImages: 'googleImages',
    googleNews: 'googleNews',
    bingSearch: 'bingSearch',
    amazonSearch: 'amazonSearch',
};
