// n8n-nodes-scraping-dog/dist/credentials/scrapingDogApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.scrapingDogApi = void 0;
class scrapingDogApi {
    constructor() {
        this.name = 'scrapingDogApi';
        this.displayName = 'ScrapingDog API';
        // Uses the link to this tutorial as an example
        // Replace with your own docs links when building your own nodes
        this.documentationUrl = 'https://docs.n8n.io/integrations/creating-nodes/build/declarative-style-node/';
        this.properties = [
            {
                displayName: 'API Key',
                name: 'apiKey',
                type: 'string',
                default: '',
                typeOptions: {
                    password: true,
                },
            },
        ];
        this.authenticate = {
            type: 'generic',
            properties: {
                qs: {
                    'api_key': '={{$credentials.apiKey}}'
                }
            },
        };
        // Credential test block
        this.test = {
            request: {
                method: 'GET',
                url: 'https://api.scrapingdog.com/scrape',
                qs: {
                    api_key: '={{$credentials.apiKey}}',
                    url: 'https://httpbin.org/get',
                },
            },
        };
    }
}
exports.scrapingDogApi = scrapingDogApi;
